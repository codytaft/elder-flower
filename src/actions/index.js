export const setCurrentUser = user => ({
  type: 'SET_CURRENT_USER',
  user
});

export const createCarer = user => ({
  type: 'CREATE_CARER',
  user
});

export const createElder = user => ({
  type: 'CREATE_ELDER',
  user
});
