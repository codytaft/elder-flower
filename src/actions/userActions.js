export const setCurrentUser = user => ({
  type: 'SET_CURRENT_USER',
  user
});

export const createUser = user => ({
  type: 'CREATE_USER',
  user
});
