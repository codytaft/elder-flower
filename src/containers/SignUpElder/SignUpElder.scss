.signup-form {
  display: flex;
  flex-direction: column;
  align-content: center;
  width: 25rem;
  margin: auto;
}

.phone-test-btn {
  display: flex;
  justify-content: center;
  align-content: center;
  width: 5rem;
  height: 4rem;
  border: 0.1rem solid #453d30;
  background-color: seashell;
  border-radius: 3rem;
  font-size: 32px;
  cursor: pointer;
}

.phone-test-btn:hover {
  background-color: rgb(215, 150, 141);
}

.phone-test-btn:focus {
  outline: none;
}

.signup-input {
  text-decoration: none;
  font-size: 32px;
  margin: 1.5rem auto 1.5rem auto;
  border-radius: 3rem;
  padding-left: 1rem;
  border: 0.1rem solid #453d30;
  color: #453d30;
  height: 4rem;
  background-color: seashell;
  width: 25rem;
}

.signup-input:focus {
  outline: none;
}

.signup-phone-section {
  display: flex;
  justify-content: space-between;
  width: 33.5rem;
  align-items: center;
}

.signup-submit-btn {
  margin: 1.5rem auto;
  text-decoration: none;
  border: 0.2rem solid rgb(78, 45, 29);
  color: #4e2d1d;
  border-radius: 5rem;
  cursor: pointer;
  width: 26rem;
  height: 4rem;
  font-size: 32px;
  background-color: seashell;
}

.signup-submit-btn:hover {
  background-color: rgb(215, 150, 141);
}

.signup-submit-btn:focus {
  outline: none;
}

.signup-submit-btn:disabled {
  opacity: 0.5;
}

.signup-submit-btn:disabled:hover {
  background-color: seashell;
}
