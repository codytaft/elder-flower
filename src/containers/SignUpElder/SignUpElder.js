import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createElder, setCurrentUser } from '../../actions';
import { withRouter } from 'react-router-dom';

import './SignUpElder.css';
export class SignUpElder extends Component {
  constructor() {
    super();
    this.state = {
      firstName: '',
      lastName: '',
      phoneNumber: '',
      emailAddress: '',
      carerName: '',
      carerPhone: ''
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    // const {
    //   firstName,
    //   lastName,
    //   phoneNumber,
    //   emailAddress,
    //   emergencyName,
    //   emergencyPhone
    // } = this.state;
    this.props.setCurrentUser(this.state);
  };

  testPhoneNumber = e => {
    e.preventDefault;
    const testPhoneNumber = this.state.phoneNumber;
  };

  render() {
    return (
      <section className="signup-page">
        <form className="signup-form" onSubmit={this.handleSubmit}>
          <input
            onChange={this.handleChange}
            className="signup-first-name"
            value={this.state.firstName}
            name="firstName"
            placeholder="First Name"
          />
          <input
            onChange={this.handleChange}
            className="signup-last-name"
            value={this.state.lastName}
            name="lastName"
            placeholder="Last Name"
          />
          <section className="signup-phone-section">
            <input
              onChange={this.handleChange}
              className="signup-phoneNumber"
              value={this.state.phoneNumber}
              name="phoneNumber"
              placeholder="Phone Number"
            />
            <button onClick={this.testPhoneNumber} className="phone-test-btn">
              Test Phone
            </button>
          </section>

          <input
            onChange={this.handleChange}
            className="signup-email-address"
            value={this.state.emailAddress}
            name="emailAddress"
            placeholder="Email Address"
          />
          <input
            onChange={this.handleChange}
            className="signup-carer-name"
            value={this.state.emergencyName}
            name="carerName"
            placeHolder="Carer Contact Name"
          />
          <input
            onChange={this.handleChange}
            className="signup-carer-phone"
            value={this.state.carerPhone}
            name="carerPhone"
            placeHolder="Carer Contact Phone"
          />
          <button className="signup-submit-btn">Submit</button>
        </form>
      </section>
    );
  }
}

export const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export const mapDispatchToProps = dispatch => ({
  createElder: user => dispatch(createElder(user)),
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(SignUpElder)
);
