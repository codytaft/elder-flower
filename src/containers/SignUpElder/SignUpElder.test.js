import React from 'react';
import { shallow } from 'enzyme';
import { SignUpElder } from './SignUpElder';
import { setCurrentUser } from '../../actions';
import { mapStateToProps, mapDispatchToProps } from './SignUpElder';
import bcrypt from 'bcryptjs';

jest.mock('bcryptjs', () => ({
  genSaltSync: () => 'salt',
  hashSync: () => 'yournewhash'
}));

describe('SignUpElder', () => {
  it('should match the snapshot', () => {
    let wrapper;
    wrapper = shallow(<SignUpElder />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('handleChange', () => {
  it('should update state', () => {
    let wrapper = shallow(<SignUpElder />);
    const mockFirstNameEvent = { target: { name: 'firstName', value: 'cody' } };
    const mockLastNameEvent = { target: { name: 'lastName', value: 'taft' } };
    const mockPhoneNumberEvent = {
      target: { name: 'phoneNumber', value: '+19038511575' }
    };
    const mockEmailEvent = {
      target: { name: 'email', value: 'cody.taft@gmail.com' }
    };
    const mockPasswordEvent = { target: { name: 'password', value: '123' } };
    const mockContactPhoneEvent = {
      target: { name: 'contactPhone', value: '+17203304593' }
    };
    const mockContactNameEvent = {
      target: { name: 'contactName', value: 'Gaynell' }
    };

    wrapper.instance().handleChange(mockFirstNameEvent);
    expect(wrapper.state('firstName')).toEqual(mockFirstNameEvent.target.value);

    wrapper.instance().handleChange(mockLastNameEvent);
    expect(wrapper.state('lastName')).toEqual(mockLastNameEvent.target.value);

    wrapper.instance().handleChange(mockPhoneNumberEvent);
    expect(wrapper.state('phoneNumber')).toEqual(
      mockPhoneNumberEvent.target.value
    );

    wrapper.instance().handleChange(mockEmailEvent);
    expect(wrapper.state('firstName')).toEqual(mockFirstNameEvent.target.value);

    wrapper.instance().handleChange(mockPasswordEvent);
    expect(wrapper.state('password')).toEqual(mockPasswordEvent.target.value);

    wrapper.instance().handleChange(mockContactPhoneEvent);
    expect(wrapper.state('contactPhone')).toEqual(
      mockContactPhoneEvent.target.value
    );

    wrapper.instance().handleChange(mockContactNameEvent);
    expect(wrapper.state('contactName')).toEqual(
      mockContactNameEvent.target.value
    );
  });
});

describe('handleSubmit', () => {
  let wrapper;
  let mockSetCurrentUser;
  let mockHistory;
  let mockCurrentUser;
  let mockUser;
  beforeEach(async () => {
    mockSetCurrentUser = await jest.fn();
    mockHistory = { push: jest.fn() };
    wrapper = shallow(
      <SignUpElder setCurrentUser={mockSetCurrentUser} history={mockHistory} />
    );

    mockUser = {
      user: {
        firstName: 'Cody',
        lastName: 'Taft',
        phoneNumber: '+19038511575',
        email: 'cody.taft@gmail.com',
        password: '123',
        contactName: 'Gaynell',
        contactPhone: '+17203304593',
        isElder: true
      }
    };
    mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
  });
  it('should set state with the correct phone numbers and password', async () => {
    let mockEvent = { preventDefault: () => jest.fn() };
    let phoneNumber = '+19038511575';
    let contactPhone = '+19038511575';
    let salt = bcrypt.genSaltSync();
    let hash = bcrypt.hashSync();
    mockUser = {
      user: {
        firstName: 'Cody',
        lastName: 'Taft',
        phoneNumber: '+19038511575',
        email: 'cody.taft@gmail.com',
        password: hash,
        contactName: 'Gaynell',
        contactPhone: '+17203304593',
        isElder: true
      }
    };
    wrapper.setState({
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '9038511575',
      email: 'cody.taft@gmail.com',
      password: hash,
      contactName: 'Gaynell',
      contactPhone: '7203304593'
    });

    let mockResult = [
      'http://localhost:3000/api/v1/users/',
      {
        method: 'POST',
        body: JSON.stringify(mockUser),
        headers: {
          'Content-Type': 'application/json'
        }
      }
    ];

    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockCurrentUser)
      })
    );

    await wrapper.instance().handleSubmit(mockEvent);
    await wrapper.setState({ phoneNumber, contactPhone, password: hash });
    expect(window.fetch).toHaveBeenCalledWith(...mockResult);
  });
});

describe('testPhoneNumber', () => {
  let wrapper;
  let mockTestResponsePhoneNumber;
  beforeEach(() => {
    mockTestResponsePhoneNumber = jest.fn();
    wrapper = shallow(
      <SignUpElder testResponsePhoneNumber={mockTestResponsePhoneNumber} />
    );
  });
  it('should invoke the testResponsePhoneNumber method', async () => {
    let mockEvent = { preventDefault: () => jest.fn() };
    let mockUser = {
      name: 'cody',
      to: '+19038511575',
      body: `If this is cody, type 'yes'. If not type 'no'.`
    };

    let mockResult = [
      'http://localhost:3000/api/sendMessage',
      {
        method: 'POST',
        body: JSON.stringify(mockUser),
        headers: {
          'Content-Type': 'application/json'
        }
      }
    ];

    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockUser)
      })
    );

    wrapper.setState({ firstName: 'cody', phoneNumber: '9038511575' });

    await wrapper.instance().testPhoneNumber(mockEvent);

    expect(window.fetch).toHaveBeenCalledWith(...mockResult);
  });
});

describe('mapStateToProps', () => {
  it('should return an object with a currentUser object', () => {
    const mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
    const mockState = {
      currentUser: { ...mockCurrentUser }
    };

    const expected = {
      currentUser: {
        id: 75,
        firstName: 'Cody',
        lastName: 'Taft',
        phoneNumber: '+19038511575',
        email: 'cody.taft@gmail.com',
        password: '123',
        contactName: 'Gaynell',
        contactPhone: '+17203304593',
        isElder: false
      }
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should invoke dispatch when using a function from mapDispatchToProps', () => {
    const mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
    const mockDispatch = jest.fn();
    const actionToDispatch = setCurrentUser(mockCurrentUser);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setCurrentUser(mockCurrentUser);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
