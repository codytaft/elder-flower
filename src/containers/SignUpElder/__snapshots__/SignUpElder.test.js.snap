// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpElder should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpElder />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="signup-first-name signup-input"
          name="firstName"
          onChange={[Function]}
          placeholder="First Name"
          value=""
/>,
        <input
          className="signup-last-name, signup-input"
          name="lastName"
          onChange={[Function]}
          placeholder="Last Name"
          value=""
/>,
        <input
          className="signup-email-address signup-input"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          value=""
/>,
        <input
          className="signup-password signup-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          value=""
/>,
        <section
          className="signup-phone-section"
>
          <input
                    className="signup-phoneNumber signup-input"
                    name="phoneNumber"
                    onChange={[Function]}
                    placeholder="Phone Number"
                    style={
                              Object {
                                        "marginLeft": "0px",
                                      }
                    }
                    value=""
          />
          <button
                    className="phone-test-btn"
                    isDisabled={true}
                    onClick={[Function]}
          >
                    Test
          </button>
</section>,
        <input
          className="signup-carer-name signup-input"
          name="contactName"
          onChange={[Function]}
          placeholder="Carer Contact Name"
          value=""
/>,
        <input
          className="signup-carer-phone signup-input"
          name="contactPhone"
          onChange={[Function]}
          placeholder="Carer Contact Phone"
          value=""
/>,
        <button
          className="signup-submit-btn"
          disabled={true}
          onClick={[Function]}
>
          Submit
</button>,
      ],
      "className": "signup-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "signup-first-name signup-input",
          "name": "firstName",
          "onChange": [Function],
          "placeholder": "First Name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "signup-last-name, signup-input",
          "name": "lastName",
          "onChange": [Function],
          "placeholder": "Last Name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "signup-email-address signup-input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email Address",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "signup-password signup-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="signup-phoneNumber signup-input"
              name="phoneNumber"
              onChange={[Function]}
              placeholder="Phone Number"
              style={
                            Object {
                                          "marginLeft": "0px",
                                        }
              }
              value=""
/>,
            <button
              className="phone-test-btn"
              isDisabled={true}
              onClick={[Function]}
>
              Test
</button>,
          ],
          "className": "signup-phone-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "signup-phoneNumber signup-input",
              "name": "phoneNumber",
              "onChange": [Function],
              "placeholder": "Phone Number",
              "style": Object {
                "marginLeft": "0px",
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "className": "phone-test-btn",
              "isDisabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Test",
            "type": "button",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "signup-carer-name signup-input",
          "name": "contactName",
          "onChange": [Function],
          "placeholder": "Carer Contact Name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "signup-carer-phone signup-input",
          "name": "contactPhone",
          "onChange": [Function],
          "placeholder": "Carer Contact Phone",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "signup-submit-btn",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="signup-first-name signup-input"
            name="firstName"
            onChange={[Function]}
            placeholder="First Name"
            value=""
/>,
          <input
            className="signup-last-name, signup-input"
            name="lastName"
            onChange={[Function]}
            placeholder="Last Name"
            value=""
/>,
          <input
            className="signup-email-address signup-input"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            value=""
/>,
          <input
            className="signup-password signup-input"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            value=""
/>,
          <section
            className="signup-phone-section"
>
            <input
                        className="signup-phoneNumber signup-input"
                        name="phoneNumber"
                        onChange={[Function]}
                        placeholder="Phone Number"
                        style={
                                    Object {
                                                "marginLeft": "0px",
                                              }
                        }
                        value=""
            />
            <button
                        className="phone-test-btn"
                        isDisabled={true}
                        onClick={[Function]}
            >
                        Test
            </button>
</section>,
          <input
            className="signup-carer-name signup-input"
            name="contactName"
            onChange={[Function]}
            placeholder="Carer Contact Name"
            value=""
/>,
          <input
            className="signup-carer-phone signup-input"
            name="contactPhone"
            onChange={[Function]}
            placeholder="Carer Contact Phone"
            value=""
/>,
          <button
            className="signup-submit-btn"
            disabled={true}
            onClick={[Function]}
>
            Submit
</button>,
        ],
        "className": "signup-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signup-first-name signup-input",
            "name": "firstName",
            "onChange": [Function],
            "placeholder": "First Name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signup-last-name, signup-input",
            "name": "lastName",
            "onChange": [Function],
            "placeholder": "Last Name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signup-email-address signup-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email Address",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signup-password signup-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="signup-phoneNumber signup-input"
                name="phoneNumber"
                onChange={[Function]}
                placeholder="Phone Number"
                style={
                                Object {
                                                "marginLeft": "0px",
                                              }
                }
                value=""
/>,
              <button
                className="phone-test-btn"
                isDisabled={true}
                onClick={[Function]}
>
                Test
</button>,
            ],
            "className": "signup-phone-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "signup-phoneNumber signup-input",
                "name": "phoneNumber",
                "onChange": [Function],
                "placeholder": "Phone Number",
                "style": Object {
                  "marginLeft": "0px",
                },
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
                "className": "phone-test-btn",
                "isDisabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Test",
              "type": "button",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signup-carer-name signup-input",
            "name": "contactName",
            "onChange": [Function],
            "placeholder": "Carer Contact Name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "signup-carer-phone signup-input",
            "name": "contactPhone",
            "onChange": [Function],
            "placeholder": "Carer Contact Phone",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "signup-submit-btn",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
